sql-plugin/src/main/301db/scala/com/nvidia/spark/rapids/shims/v2/GpuBroadcastHashJoinExec.scala

```diff
@@ -69,22 +114,40 @@ class GpuBroadcastHashJoinMeta(
 
   override def convertToGpu(): GpuExec = {
     val Seq(left, right) = childPlans.map(_.convertIfNeeded())
+    val gpuBuildSide = GpuJoinUtils.getGpuBuildSide(join.buildSide)
     // The broadcast part of this must be a BroadcastExchangeExec
     val buildSide = join.buildSide match {
       case BuildLeft => left
       case BuildRight => right
     }
     verifyBuildSideWasReplaced(buildSide)
-    val joinExec = GpuBroadcastHashJoinExec(
-      leftKeys.map(_.convertToGpu()),
-      rightKeys.map(_.convertToGpu()),
-      join.joinType,
-      GpuJoinUtils.getGpuBuildSide(join.buildSide),
-      None,
-      left, right)
-    // The GPU does not yet support conditional joins, so conditions are implemented
-    // as a filter after the join when possible.
-    condition.map(c => GpuFilterExec(c.convertToGpu(), joinExec)).getOrElse(joinExec)
+
+    val condition = conditionMeta.map(_.convertToGpu())
```

Comment:
should `convertToGpu` happen if and only if this is the correct type of join and `isAstCondition`? In other words, should it happen given `substituteBroadcastNestedLoopJoin` is true?

Reply:
I think it can be.  I will remove the local val and move it to where it's used.